tests/jballerina-unit-test/src/test/resources/test-src/expressions/async/basic-async-operations.bal

```diff
@@ -90,12 +91,18 @@ function testAsyncNonNativeBasic8() returns int {
     return result;
 }
 
-function testAsyncNonNativeBasic9() returns int {
```

Comment:
Why this test changed?

Reply:
For alternative wait the expected behavior depends on the non-error return value. In this test case, `future f1` is cancelled
In this test, the alternate wait panics with future cancelled error as we manually cancel the f1 before the wait.
Since it's a panic the wait should terminate as per the spec. 
I moved the assertion to Ballerina side for handling the error.
And I added the if condition as there may be a chance of f2 completing before which will result in 7.